### PR TITLE
[#40228] Added new endpoint (linkable work packages for a storage)

### PR DESCRIPTION
**Pull request by Kharonus**
- https://community.openproject.org/work_packages/40228
- new endpoint for getting all linkable work packages for a storage

### CODE REVIEW

File: docs/api/apiv3/components/schemas/work_packages_model.yml [hunk 1]
```diff
@@ -1,6 +1,43 @@
...
+            readOnly: true
+  _embedded:
+    type: object
+    required:
+      - elements
+    properties:
+      elements:
+        type: array
+        items:
+          $ref: '#/components/schemas/Work_PackageModel'
```
Comment (wielinde) [inline docs/api/apiv3/components/schemas/work_packages_model.yml:39]: A bit unrelated: I just came across this line and found `Work_PackageModel` to be very ugly. Is there a need of mixing camel case and snake case? Here Work and PackageModel look like two different words. Can we drop upcase in file names?

File: docs/api/apiv3/paths/storage_linkable_work_packages.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# /api/v3/storages/{id}/linkable_work_packages
```
Comment (wielinde) [inline docs/api/apiv3/paths/storage_linkable_work_packages.yml:1]: I argue that we do not need the storage in the path. What value does it bring? In the end we return work packages that are linkable, no matter from which storage provider we send the request from. This has the great benefit of removing the storage ID from the path. The ID is currently not easy to obtain. // @Kharonus @ulferts 

```suggestion
# /api/v3/linkable_work_packages
```

Comment (wielinde) [reply docs/api/apiv3/paths/storage_linkable_work_packages.yml:1]: I am wrong. Of cause we need to provide the storage as we only can link to those work package in projects that have the module enabled AND the storage selected for the project.

Still it probably is not {id} but {host}

Comment (Kharonus) [reply docs/api/apiv3/paths/storage_linkable_work_packages.yml:1]: How do we solve the use case when we only want to return work packages that are explicitly linkable for the current storage I'm asking for? should we url encode the host and put it into a filter? or switch the request to a post to use a request body?

Comment (wielinde) [reply docs/api/apiv3/paths/storage_linkable_work_packages.yml:1]: Good point. I'd say a filter accepting the url-encoded host as a parameter.

Comment (Kharonus) [reply docs/api/apiv3/paths/storage_linkable_work_packages.yml:1]: Alright, then the next question would be, if we want a separate endpoint or just add the filter to the work packages endpoint?

Mind game: if we do not provide a filter, would the collection differ from what is returned from the standard work package endpoint?

File: docs/api/apiv3/components/schemas/work_packages_model.yml (hunk 1 above)
Comment (Kharonus) [reply docs/api/apiv3/components/schemas/work_packages_model.yml:39]: Well, I agree them to be ugly. I didn't want to change all of them at once. And, like always for naming: We should have an agreement in dev team, how we want them to be.

File: docs/api/apiv3/paths/storage_linkable_work_packages.yml (hunk 2 above)
Comment (wielinde) [reply docs/api/apiv3/paths/storage_linkable_work_packages.yml:1]: Good point. So without a filter `linkable_storage` the endpoint would not be  different to the standard work packages endpoint. So maybe we don't create a new endpoint at all and only create a filter for the work packages query.

**Comment by Kharonus** [comment]
Replaced by #10202

**Pull request closed without merging**